### DISCUSSION TITLE
GMAIL RFC 5322 Problem with send BCC's

### DISCUSSION THREAD

**Discussion by acedrummond** [PHPMailer/PHPMailer#2875] [Q&A]
I am sending a newsletter to a list of 300+/- people.

I have been doing this for a long time sending to each of the recipients, sleeping for 6 seconds between each and it takes a long time.
The sleep(6) is needed so that the hosting mail server is not overrun and hosting company stops processing. This takes about 40 minutes +/- assuming there are no hiccups along the way.

What I am trying to do now is use $mail->addBcc (loading it by looping thru my list) and when I do so any traffic going to a gmail.com address is creating a delivery event failure (message not delivered) "Message is not RFC 5322 compliant" - message blocked.

All other messages in my list get thru.

I do specify an $mail->addAddress that is the same as the $mail->setFrom

**Is there a way to get gmail to accept this email that is to undisclosed recipients because I am using BCC's?**

Are there other email providers with such restrictions? or are they planned in future?

My code:

	$mail = new PHPMailer;

	$mail->setFrom('anybody@xyz.org', 'XYZ');
	$timestamp = date('Y-m-d');

        $mail->clearAddresses();

	$mail->isHTML(true);
	
	$sql ="select * from news_users where news_status = 'A' order by news_name";
        $result = $db->query($sql);

        // Add BCC from database
	    $mail_count = 0;
	    while ( $db_field = mysqli_fetch_assoc($result) )
	          {
	           $member_name  = $db_field['news_name'];
	           $email                 = $db_field['news_email'];
	           $news_seq          = $db_field['seq'];
	           
	           $mail->addBcc($email,$member_name);
	           $mail_count = $mail_count + 1; 
	         }          
    // Sending copy any email sent to newsletter editor!     
    $mail->addAddress('anybody@xyz.org', 'XYZ'); 
    
    $mail->Subject = "VRCS NEWSLETTER Now Available";

    // Body of the text either can contain default primary message
    // replacement primary message and possibly additional message
        		
	$mail->Body = $msg_text.$additional_text;

    // Alt Body for those who cant tolerate HTML
    
    $this_msg = $msg_text;
    $this_msg = strip_tags($this_msg, "<b>");
    $mail->AltBody = $this_msg;

    $mail->AltBody = $this_msg;
   
	if(!$mail->send())
	  {
	   echo "Mailer Error: " . $mail->ErrorInfo;
	   exit;
	  }
	else
	  {
	   echo "Message has been sent successfully<br>";
	  }

**Comment by Synchro** [comment]
I suspect your issue is that the php mail function might be mishandling the `To` header generation in the absence of named recipients, resulting in duplicate headers, causing the RFC5322 issue.

I suggest trying to use SMTP to localhost instead, which you should be able to do by adding this line:

    $mail->isSMTP();

**Reply by acedrummond** [reply]
Thanks for your input.

See error message returned to host system.

It complains about 'duplicates'. (see attached)

If I send my list as a CC and not BCC it works properly. The BCC get addressed as 'undisclosed list' and that is the issue I am battling.

I just tried send via php mail function (STMP) instead and get the same results. It is gmails restriction as far as I can tell.

According to what I am reading RFC 5322 requires a To: to match the sender and it should work, but so far no luck.

[delivery problem.pdf](https://github.com/PHPMailer/PHPMailer/files/10817191/delivery.problem.pdf)
